src/esssans/loki/io.py

```diff
@@ -87,46 +106,94 @@ def _merge_runs(data_groups: Iterable[sc.DataGroup], entries: Iterable[str]):
     return out
 
 
+def merge_sample_runs(
```

Comment:
I had to specialise the types here for `SampleRun` and `BackgroundRun`, instead of having a single function with `ScatteringRunType`, because if not there would be multiple providers for `LoadedFileContents[RunType]` as the `to_file_contents` below also matches.

I don't think it's pretty. We also need to manually insert the `merge_sample_runs` and `merge_background_runs` into the pipeline if we want to use a parameter table with multiple runs.

Is there a better way to achieve this?
I basically want to merge the runs if a parameter table is supplied, but do nothing otherwise.

Reply:
We can ignore this problem for now, as we plan to have a more generic solution.